File: frontend/src/types/project.ts
/** 阶段状态 */
export interface StageStatuses {
  import_status: "pending" | "completed" | "dirty";
  normalize_status: "pending" | "completed" | "dirty";
  grouping_status: "pending" | "completed" | "dirty";
  compliance_status: "skipped" | "pending" | "completed" | "dirty";
  comparison_status: "pending" | "completed" | "dirty";
}

/** 项目列表项 */
export interface ProjectSummary {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
  supplier_count: number;
  current_stage: string;
}

/** 项目详情 */
export interface ProjectDetail {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
  stage_statuses: StageStatuses;
}

/** 创建项目请求 */
export interface CreateProjectRequest {
  name: string;
}

File: backend/models/project.py
from pydantic import BaseModel, Field


class StageStatuses(BaseModel):
    import_status: str = "pending"  # pending | completed | dirty
    normalize_status: str = "pending"
    grouping_status: str = "pending"
    compliance_status: str = "skipped"  # skipped | pending | completed | dirty
    comparison_status: str = "pending"


class ProjectCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)


class ProjectSummary(BaseModel):
    id: str
    name: str
    created_at: str
    updated_at: str
    supplier_count: int = 0
    current_stage: str = "导入文件"


class ProjectDetail(BaseModel):
    id: str
    name: str
    created_at: str
    updated_at: str
    stage_statuses: StageStatuses
